datacube/drivers/postgis/alembic/versions/d27eed82e1f6_remove_datasetlocation_table.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of the Open Data Cube, see https://opendatacube.org for more information
+#
+# Copyright (c) 2015-2024 ODC Contributors
+# SPDX-License-Identifier: Apache-2.0
+"""remove DatasetLocation table
+
+Revision ID: d27eed82e1f6
+Revises: 610f32dca3cb
+Create Date: 2024-11-13 02:39:55.671819
+
+"""
+from typing import Sequence, Union
+
+from alembic import op
+from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey, UniqueConstraint, Index
+from sqlalchemy.dialects import postgresql as postgres
+from sqlalchemy.sql import func
+from datacube.drivers.postgis._core import METADATA
+
+
+# revision identifiers, used by Alembic.
+revision: str = 'd27eed82e1f6'
+down_revision: Union[str, None] = '610f32dca3cb'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.add_column("dataset",
+                  Column("uri_scheme", String, comment="The scheme of the uri."),
+                  schema="odc")
+    op.add_column("dataset",
+                  Column("uri_body", String, comment="The body of the uri."),
+                  schema="odc")
```

Comment:
After adding the new columns and before dropping the old table - you should copy the existing locations from the old table to the new column.

Migrations are for migrating databases, not just schemas.